### ISSUE TITLE
Using "unitless zero" for <length> leads to mistakenly report value is invalid

### ISSUE THREAD

**Issue by limonte** [csstree/csstree#103]
I'm using the latest (1.5.0) `stylelint-csstree-validator` and this line:

```css
background: url('image.png') 0 30px no-repeat;
 ```

produces an error:

```
Invalid value for `background`   csstree/validator
```

Changing it to

```css
background: url('image.png') 0px 30px no-repeat;
 ```

fixes the `csstree/validator` issue, but produces a new one:

```
Unexpected unit                  length-zero-no-unit
```

So, the latest `stylelint-csstree-validator` is in conflict with the `length-zero-no-unit` rule.

**Comment by lahmatiy** [comment]
Looks like there is an issue with low priority matching I didn't caught during testing. Will take a look. Thank you for report!

**Comment by lahmatiy** [comment]
I localized the problem, it happens on syntaxes with long (more than 5 terms) `||-` or `&&-` groups. I started refactoring a code that responsible for it and problem has gone :)
Matching after a refactoring:
![image](https://user-images.githubusercontent.com/270491/61057017-1e2f9900-a3fd-11e9-96d8-603e9456dd60.png)

Also matching takes less interaction count than before (match result before patch):
![image](https://user-images.githubusercontent.com/270491/61057327-bf1e5400-a3fd-11e9-886c-42cfc124268e.png)

**Comment by lahmatiy** [comment]
Just published patch versions of `css-tree` (1.0.0-alpha.32) and `stylelint-csstree-validator` (1.5.1)

**Comment by limonte** [comment]
That was quick, thanks a lot!

**Comment by julianaleon** [comment]
I think I'm running into the same issue with this bit of CSS:
`box-shadow: inset 0 1px 3px rgba(0, 0, 0, .1);`

It's throwing this error:
<img width="520" alt="Screen Shot 2019-07-11 at 11 53 08 AM" src="https://user-images.githubusercontent.com/6567368/61069734-9765c680-a3d2-11e9-979b-34435d3f3f9f.png">

I tried bumping to v1.5.1 but the problem still persists.

**Comment by lahmatiy** [comment]
@julianaleon Yep, the same sort of problem but another issue. Will take a look asap.

**Comment by lahmatiy** [comment]
Finally, I found a simple and robust (I believe) solution.
Just released `css-tree` (1.0.0-alpha.33) and `stylelint-csstree-validator` (1.5.2) with the fix
